src/olympia/addons/tests/test_views.py

```diff
@@ -2629,6 +2629,25 @@ def test_bad_filter(self):
             self.url, {'app': 'lol'}, expected_status=400)
         assert data == ['Invalid "app" parameter.']
 
+    def test_filter_by_author(self):
+        author = user_factory(username=u'my-FancyAuthorNâme')
+        addon = addon_factory(slug='my-addon', name=u'My Addôn',
+                              tags=['some_tag'], weekly_downloads=999)
+        addon2 = addon_factory(slug='another-addon', name=u'Another Addôn',
+                               tags=['unique_tag', 'some_tag'],
+                               weekly_downloads=333)
+        AddonUser.objects.create(addon=addon, user=author)
+        AddonUser.objects.create(addon=addon2, user=user_factory())
+        self.reindex(Addon)
+
+        data = self.perform_search(self.url, {'author': u'my-FancyAuthorNâme'})
+        assert data['count'] == 1
+        assert len(data['results']) == 1
+
+        result = data['results'][0]
+        assert result['id'] == addon.pk
+        assert result['slug'] == addon.slug
```

Comment:
I'm not sure if we should have fuzzy matching for this too, not sure how useful a name search by exact author name is in real-life.

In any case, this test needs to verify either that we don't support fuzzy search, or that we do and test that.